### ISSUE TITLE
Interchange and generated rules

### ISSUE THREAD

**Issue by bredele** [zurb/foundation-sites#8335]
Hi,

My team and I would like to use the interchange plugin but we noticed an error in the `[generatedRules](https://github.com/zurb/foundation-sites/blob/develop/js/foundation.interchange.js#L104)` handler. Here's our HTML code:

```html
<img data-interchange="[/path/to/bigger-image.jpg, (large)]">
```

This method is trying to loop over an array using the `for in` loop and so, slice some of the array prototype functions.

Thanks for your help.

**Comment by bredele** [comment]
Just some extra information, we (unfortunately) use extjs which overrides some native methods of the Array prototype and add other methods such as `remove`.

**Comment by jyoakum-swx** [comment]
I had the same issue with a similar library.
I was able to get this working for me by wrapping the contents of 'for each' blocks with a hasOwnProperty check. 

```
    for (var i in rules) {
      if (rules.hasOwnProperty(i)) {
          // original code
      }
    }
```
Note: There are several 'for each' blocks in the script.

**Comment by colin-marshall** [comment]
Closed in #8686. Thanks @jyoakum-swx for the solution!
